core/storage/feedback/gae_models.py

```diff
@@ -344,6 +382,72 @@ def get_all_messages(cls, page_size, urlsafe_start_cursor):
             cls.query(), page_size, urlsafe_start_cursor)
 
 
+class FeedbackThreadUserModel(base_models.BaseModel):
+    """Model for storing the ids of the messages in the thread that are read by
+    the user.
+
+    Instances of this class have keys of the form [user_id].[thread_id]
+    """
+    message_ids_read_by_user = ndb.IntegerProperty(repeated=True, indexed=True)
+
+    @classmethod
+    def generate_full_id(cls, user_id, exploration_id, thread_id):
```

Comment:
I did not keep this a private method as I would need it to generate ids outside the class. This is similar to the 'generate_full_thread_id' method of FeedbackThreadModel.